fineract-provider/src/main/java/org/apache/fineract/notification/domain/NotificationGenerator.java

```diff
@@ -0,0 +1,178 @@
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "notification_generator")
+public class NotificationGenerator extends AbstractPersistableCustom<Long> {
+
+    @Column(name = "object_type")
+    private String objectType;
+
+    @Column(name = "object_identifier")
+    private Long objectIdentifier;
+
+    @Column(name = "action")
+    private String action;
+
+    @Column(name = "actor")
+    private String actor;
+
+    @Column(name = "is_system_generated")
+    private boolean isSystemGenerated;
+
+    @Column(name = "notification_content")
+    private String notificationContent;
+
+    @Column(name = "created_at")
+    private String createdAt;
+
+    public NotificationGenerator() {}
+
+    public NotificationGenerator(NotificationGeneratorBuilder builder) {
+        this.objectType = builder.objectType;
+        this.objectIdentifier = builder.objectIdentifier;
+        this.action = builder.action;
+        this.actor = builder.actor;
+        this.isSystemGenerated = builder.isSystemGenerated;
+        this.notificationContent = builder.notificationContent;
+        this.createdAt = builder.createdAt;
+    }
+
+    public String getObjectType() {
+        return objectType;
+    }
+
+    public void setObjectType(String objectType) {
+        this.objectType = objectType;
+    }
+
+    public Long getObjectIdentifier() {
+        return objectIdentifier;
+    }
+
+    public void setObjectIdentifier(Long objectIdentifier) {
+        this.objectIdentifier = objectIdentifier;
+    }
+
+    public String getAction() {
+        return action;
+    }
+
+    public void setAction(String action) {
+        this.action = action;
+    }
+
+    public String getActor() {
+        return actor;
+    }
+
+    public void setActor(String actor) {
+        this.actor = actor;
+    }
+
+    public boolean isSystemGenerated() {
+        return isSystemGenerated;
+    }
+
+    public void setSystemGenerated(boolean systemGenerated) {
+        isSystemGenerated = systemGenerated;
+    }
+
+    public String getNotificationContent() {
+        return notificationContent;
+    }
+
+    public void setNotificationContent(String notificationContent) {
+        this.notificationContent = notificationContent;
+    }
+
+
+    @Override
+    public String toString() {
+        return "NotificationGenerator{" +
+                "id=" + getId() +
+                ", objectType='" + objectType + '\'' +
+                ", objectIdentifier=" + objectIdentifier +
+                ", action='" + action + '\'' +
+                ", actor='" + actor + '\'' +
+                ", isSystemGenerated=" + isSystemGenerated +
+                ", notificationContent='" + notificationContent + '\'' +
+                ", createdAt=" + createdAt +
+                '}';
+    }
+
+    public static class NotificationGeneratorBuilder {
```

Comment:
NotificationGeneratorBuilder should be separate java file as NotificationGenerator should follow SRP? NotificationGenerator name will confuse the users as it is not a generator but it is Entity.